### DISCUSSION TITLE
Blank Oled on power cycle/ reset

### DISCUSSION THREAD

**Discussion by hackintoshlover** [davetcc/tcMenu#166] [Q&A] [answered]
Hello everyone,
I am using a 0.91" oled and arduino mega with tcmenu to create a custom interface.
I am encountering a strange problem.
Basically when i upload the code everything works fine, bt as soon as i power cycle or reset the board, oled goes blank, even re-uploading the code does not help.
To get it working back again i need to upload any example from u*g2 first and then re-upload the tcmenu code and it works until powerycle.
This problem happens only while using tcmenu code.
Any other examle apart from tcmenu code works perfectly fine even after reset or power cycle.
Kindly help.

**Comment by davetcc** [comment]
Converted to a discussion. 

Can you please provide the versions of everything you're using here (if you are using designer 2.2, there is a copy diagnostics option on the help menu). Also include the hardware that you are targetting too please.

Also, if you have the lower latency mode checked, can you please check if you have Wire.begin(..) with the right parameters for your board (if any) before calling setupMenu(). We will make this clearer in the documentation, and produce a note during the code generator that this is needed.

**Reply by hackintoshlover** [reply]
**Hardware:**
Arduino Mega, Rotary Encoder, 0.91" oled display
**Software:** 
TcMenuDesigner Version - 2.2.3
Plugin versions:
core-display - 2.2.3
core-remote - 2.2.3
core-themes - 2.2.3
Library versions:
tcMenu - 2.2.3
LiquidCrystalIO - 1.4.1
TaskManagerIO - 1.2.3
IoAbstraction - 2.0.5
Arduino IDE: 1.8.13

I solved the issue by changing **WANT_TASK_MANAGER_FRIENDLY_YIELD** to 0 from 1 in **tcMenuU8g2.h**

**Reply by davetcc** [reply]
You don't need to change that flag manually. You can just turn off the latency option in the code generator U8G2 properties.

However, I suspect all that was needed to use the low latency wire was to add Wire.begin somewhere. We are doing a patch release to turn off low latency by default for beginners.

**Comment by hackintoshlover** [comment]
I have a very weird error though.
When i select AnalogMenuItem all the values in it gets highlighted.
Refer the video link attached:
https://drive.google.com/file/d/1mUl3DP927agoknIu9PBuDwSQhajta7e9/view?usp=drivesdk
How to rectify this?

**Comment by davetcc** [comment]
It looks like for some reason U8G2 is drawing in XOR mode rather than the usual mode. Please could you share the U8G2 and theme settings that you are using in the code generator? I have a very similar setup, I'll try the same settings and see if I can recreate it.

**Comment by hackintoshlover** [comment]
This is my u8g2 setting
![Screenshot (71)](https://user-images.githubusercontent.com/40655406/135718022-eb265fbf-b36d-40b3-a384-b3f88e5b370d.png)
![Screenshot (70)](https://user-images.githubusercontent.com/40655406/135718024-a4f26117-22c9-4e9e-91e1-afdbea0c853c.png)

Since i am using a 128x32 bit oled screen and in configurator that setting is not present refer below:
![Screenshot (72)](https://user-images.githubusercontent.com/40655406/135718087-e82c2f6a-3dc9-4212-8092-129b3efd3b89.png)
As you can see ssd1306 is not present in menu instead it shows ssd1305 which gives compiler warning.
Hence, in the arduino code code i have manually added **U8G2_SSD1306_128X32_UNIVISION_F_HW_I2C** this line in menu.cpp and menu.h

**Reply by davetcc** [reply]
If the display setting you need is not available in quick-start, you use manual U8G2 configuration instead, never adjust the menu generated files directly. The manual option allows you to create and start the U8G2 display yourself. IE you create the global variable, and then you initialise it before calling `setupMenu()` in `setup()`

**Comment by hackintoshlover** [comment]
I am not using any theme
![Screenshot (73)](https://user-images.githubusercontent.com/40655406/135718385-733ada24-c3c4-41fb-b060-cfb489124be1.png)
I hope its not necessary
Also one extra question:
How do i display a custom logo/image on bootup
I tried adding some functions in main loop but it conflicts so does not work

**Reply by davetcc** [reply]
To be honest you'll need to read the documentation through, especially around taking over the display and giving it back to tcMenu.

See:

https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/

and in particular 

https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/renderer-take-over-display/

**Comment by davetcc** [comment]
In terms of the strange rendering behaviour, we'll need to try and recreate it locally first, which theme have you chosen, and what are the theme settings?

**Reply by davetcc** [reply]
Just seen you are not using a theme, can you try either the mono bordered, or the mono inverse title theme.

**Reply by hackintoshlover** [reply]
Yes used a theme now everything is fine
Can you guide me on how to get a bootup logo/image?

**Reply by hackintoshlover** [reply]
After selecting a theme i noticed one thing, previously whichever menu was selected used to get highlighted but now it just shows an arrow.
Any idea how to get that back?

**Reply by davetcc** [reply]
You can change the selected colours on the theme how you like, there are two starting points, mono bordered theme, and mono inverted theme, from this point you can adjust the theme header file that tcMenu adds to your sketch.

**Reply by davetcc** [reply]
To be more complete this function in the theme header file controls the selected colors:

      factory.setSelectedColors(bg, fg);

The choices are 0 - BLACK, 1 - WHITE, 2 - XOR

**Comment by davetcc** [comment]
> Can you guide me on how to get a bootup logo/image?

If you mean taking over the display from tcMenu, then you can read about how to do that in the replies above, there's a documentation link for that. Copied again here:

https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/renderer-take-over-display/

The only thing you need to make sure of is that the setupMenu() call has been made before taking over the display.

**Reply by hackintoshlover** [reply]
Thanks i will follow this and get back.

**Reply by hackintoshlover** [reply]
I had a look at this, what i need to achieve is when the arduino boots in the setup menu it shows the image/logo, then in setup it goes to void loop and behaves normal.
The logo thing happens only once when the arduino is power cycled or reset.

**Reply by davetcc** [reply]
There are many ways you could do this using task manager, or otherwise, but it's beyond the scope of the forum to be honest.

**Reply by hackintoshlover** [reply]
Can you provide me a documentation for same?

**Reply by hackintoshlover** [reply]
Final question i wanted to increase values by 0.1 but in tcmenu designer float is not present. Any help there?

**Reply by hackintoshlover** [reply]
Also i am getting a weird error my minimum value is 865 and maximum is 867 but the menu does not stop at 867 it keeps increasing
![Screenshot (74)](https://user-images.githubusercontent.com/40655406/135723196-a1720835-88ca-4011-ab65-f5e1b6d4b9b7.png)

**Reply by davetcc** [reply]
I think you've misunderstood how analog items work, there's a guide to them here: https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/menu-item-types/analog-menu-item/

EDIT - In short, the value  is applied after the offset. So lets say you wanted a range of  100 - 110. The offset would be 100, the maximum would be 10.

**Reply by davetcc** [reply]
marking as answered for now, please reopen if you think otherwise.

**Comment by davetcc** [comment, marked as answer]
I think you've misunderstood how analog items work, there's a guide to them here: https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/menu-item-types/analog-menu-item/

EDIT - In short, the value is applied after the offset. So lets say you wanted a range of 100 - 110. The offset would be 100, the maximum would be 10.
